Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,467 @@
+# coding=utf-8
+"""
+Written by Egbert Bouman, Mihai Capotă, Elric Milon, and Ardhi Putra Pratama H
+Supported boosting sources
+"""
+import HTMLParser
+import glob
+import logging
+import os
+import random
+import time
+import urllib
+from binascii import hexlify
+from hashlib import sha1
+
+import libtorrent as lt
+from twisted.internet import reactor
+from twisted.internet.task import LoopingCall
+from twisted.web.client import Agent, readBody
+from twisted.web.http_headers import Headers
+
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.simpledefs import NTFY_INSERT, NTFY_TORRENTS, NTFY_UPDATE
+from Tribler.Core.version import version_id
+from Tribler.Main.Utility.GuiDBHandler import startWorker
+from Tribler.Main.Utility.GuiDBTuples import Torrent
+from Tribler.Main.vwxGUI.GuiUtility import GUIUtility
+from Tribler.community.allchannel.community import AllChannelCommunity
+from Tribler.community.channel.community import ChannelCommunity
+from Tribler.dispersy.exception import CommunityNotFoundException
+from Tribler.dispersy.util import call_on_reactor_thread
+
+
+class BoostingSource(object):
+    """
+    Base class for boosting source. For now, it can be RSS, directory, and channel
+    """
+
+    def __init__(self, session, source, boost_settings, callback):
+        self.session = session
+        self.channelcast_db = session.lm.channelcast_db
+
+        self.torrents = {}
+        self.source = source
+        self.interval = boost_settings.source_interval
+        self.max_torrents = boost_settings.max_torrents_per_source
+        self.callback = callback
+        self.archive = False
+
+        self.enabled = True
+
+        self.av_uprate = 0
+        self.av_dwnrate = 0
+        self.storage_used = 0
+        self.ready = False
+
+        self.gui_util = GUIUtility.getInstance()
```

Comment:
Some of method I used are indeed relatively simple, however, some of them are quite complicated. For example in `loadTorrent` in `TorrentManager`. I need to add the download state of a particular torrent to `LibraryManager` where I think all the download state in a session stored in there. If I don't use this class, wouldn't it be out of sync of Tribler torrent's downloading? The same issue happens in `loadTorrent` method where it calls other method.

I do agree to not use any GUI-related stuff, but unfortunately some function I need are there in the GUI packages. Do you think I need to re-implement them in my class?